src/jit/ee_il_dll.cpp

```diff
@@ -58,6 +58,13 @@ extern "C" void __stdcall jitStartup(ICorJitHost* jitHost)
 {
     if (g_jitInitialized)
     {
+        if (jitHost != g_jitHost)
+        {
+            // Reset JitConfig values for SuperPMI replay.
+            JitConfig.destroy(g_jitHost);
```

Comment:
IMO, this case warrants a more verbose comment. Something like:

// We normally don't expect jitStartup() to be invoked more than once. (We check whether it has been called once due to an abundance of caution.)
// However, during SuperPMI playback, we need to JIT many different functions. Each one carries its own environment configuration state.
// So, we need the JIT to reload the JitConfig state for each change in the environment state of the replayed compilations.
// We do this by calling jitStartup with a different ICorJitHost, and have the JIT re-initialize its JitConfig state when this happens.

Reply:
Thanks, fixed.